basic_module/distribution/src/types.rs

```diff
@@ -13,3 +13,115 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use ckey::{Ed25519Public as Public, NetworkId};
+use ctypes::BlockHash;
+use std::collections::{BTreeMap, HashMap};
+use std::convert::TryFrom;
+pub type BlockNumber = u64;
+
+/// Uniquely identifies block.
+#[derive(Debug, PartialEq, Copy, Clone, Hash, Eq)]
+pub enum BlockId {
+    /// Block's blake256.
+    /// Querying by hash is always faster.
+    Hash(BlockHash),
+    /// Block number within canon blockchain.
+    Number(BlockNumber),
+    /// Earliest block (genesis).
+    Earliest,
+    /// Latest mined block.
+    Latest,
+    /// Parent of latest mined block.
+    ParentOfLatest,
+}
+
+impl From<BlockHash> for BlockId {
+    fn from(hash: BlockHash) -> Self {
+        BlockId::Hash(hash)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Transaction {
+    /// Seq.
+    pub seq: u64,
+    /// Quantity of CCC to be paid as a cost for distributing this transaction to the network.
+    pub fee: u64,
+    /// Network Id
+    pub network_id: NetworkId,
+
+    pub action: Action,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Action {
+    CloseBlock {
+        total_reward: u64,
+        total_min_fee: u64,
+        stakes: HashMap<Public, u64>,
+    },
+    OpenBlock {
+        block_number: u64,
+        block_author: Public,
+        block_hash: BlockHash,
+        rewards: BTreeMap<Public, u64>,
+    },
+}
+
+const MAX_VALIDATOR_SIZE: usize = 800;
```

Comment:
Currently, it is fine. Later, we're going to utilize `foundry / types` crate to share types between host and modules.